to.etc.domui/src/main/java/to/etc/domui/util/asyncdialog/AsyncDialog.java

```diff
@@ -44,24 +62,33 @@ private AsyncDialog() {
 						}
 					}
 				} else {
+					if(null == errorException && cancelled) {
```

Comment:
That is wrong. If you cancel an operation but it finishes before it can be cancelled then it is finished, not cancelled.
A cancelled operation MUST by itself throw the cancellation exception if it cancels.

Reply:
And those ifs are really messy, but I think they will mostly disappear again.

Reply:
actually issue that I wanted to solve was that if I cancel, and it was not even started, I have no distinction from case when it successfully completed. I think if it completes  before it handles the cancel, it actually completes, and that cancelled here is false.

Reply:
Agreed to temporarily leave this change out. We need different solution for this.